spring-integration-jms/src/main/java/org/springframework/integration/jms/JmsOutboundGateway.java

```diff
@@ -1500,12 +1504,12 @@ public void run() {
 			try {
 				if (System.currentTimeMillis() - JmsOutboundGateway.this.lastSend >
 						JmsOutboundGateway.this.idleReplyContainerTimeout
-						&& JmsOutboundGateway.this.replies.size() == 0 &&
-						JmsOutboundGateway.this.replyContainer.isRunning()) {
+						&& JmsOutboundGateway.this.replies.isEmpty() &&
+						Objects.requireNonNull(JmsOutboundGateway.this.replyContainer).isRunning()) {
```

Comment:
Looks like better to propagate `this.replyContainer` as this class ctor argument from that `handleRequestMessage()`.